### PR TITLE
feat(gulp): Add gulp task mkdir:upload to ensure upload directory exi…

### PR DESCRIPTION
**Pull request by rhutchison**
…sts.

Related #1175

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -202,6 +203,11 @@ gulp.task('imagemin', function () {
     .pipe(gulp.dest('public/dist/img'));
 });
 
+// Make sure upload directory exists
+gulp.task('mkdir:upload', function () {
+  return fs.mkdir('modules/users/client/img/profile/uploads', function () { });
```
Comment (ilanbiala) [inline gulpfile.js]: @rhutchison need to handle a possible error here.

File: gulpfile.js [hunk 2]
```diff
@@ -202,6 +203,11 @@ gulp.task('imagemin', function () {
     .pipe(gulp.dest('public/dist/img'));
 });
 
+// Make sure upload directory exists
+gulp.task('mkdir:upload', function () {
```
Comment (ilanbiala) [inline gulpfile.js]: I think the name doesn't need to be the same, thoughts on maybe `mkUploadDir`?

File: gulpfile.js (hunk 1 above)
Comment (rhutchison) [inline gulpfile.js]: Error is handled with noop

Comment (ilanbiala) [inline gulpfile.js]: log it or something maybe? We basically need to notify users of it at the least.

File: gulpfile.js (hunk 2 above)
Comment (codydaig) [inline gulpfile.js]: I hesitate to change since this is what it currently is.

Comment (ilanbiala) [inline gulpfile.js]: I know, but Grunt's convention so to use different subtasks, where was can just create a proper task name in Gulp. I feel like we shouldn't keep old conventions.

File: gulpfile.js [hunk 3]
```diff
@@ -218,6 +218,15 @@ gulp.task('copyLocalEnvConfig', function () {
     .pipe(gulp.dest('config/env'));
 });
 
+// Make sure upload directory exists
+gulp.task('makeUploadsDir', function () {
+  return fs.mkdir('modules/users/client/img/profile/uploads', function (res) {
```
Comment (ilanbiala) [inline gulpfile.js]: nitpick, res should be err (convention)

**Comment by ilanbiala** [comment]
Also, the scope of the commit message should be gulp, because this is gulp-related, not really "core".

File: gulpfile.js (hunk 3 above)
Comment (rhutchison) [inline gulpfile.js]: Where is this convention defined?

This is a callback method for success/error and is not specifically a callback only for errors.

**Comment by rhutchison** [comment]
@ilanbiala scope is updated to reflect your feedback

File: gulpfile.js (hunk 3 above)
Comment (ilanbiala) [inline gulpfile.js]: It's a common node convention for callbacks to return an error, so err makes more sense.

**Comment by rhutchison** [comment]
@ilanbiala LGTY?

**Comment by rhutchison** [comment]
Closing/Opening to produce clean build.  One node timed out.

**Pull request merged**